app/views/RoomMembersView/index.js

```diff
@@ -94,19 +96,36 @@ class RoomMembersView extends React.Component {
 
 		const { room } = this.state;
 		const {
-			muteUserPermission, setLeaderPermission, setOwnerPermission, setModeratorPermission, removeUserPermission
+			muteUserPermission, setLeaderPermission, setOwnerPermission, setModeratorPermission, removeUserPermission, editTeamMemberPermission
 		} = this.props;
-		const result = await RocketChat.hasPermission([
-			muteUserPermission, setLeaderPermission, setOwnerPermission, setModeratorPermission, removeUserPermission
-		], room.rid);
-
-		this.permissions = {
-			[PERMISSION_MUTE_USER]: result[0],
-			[PERMISSION_SET_LEADER]: result[1],
-			[PERMISSION_SET_OWNER]: result[2],
-			[PERMISSION_SET_MODERATOR]: result[3],
-			[PERMISSION_REMOVE_USER]: result[4]
-		};
+		let result;
+
+		if (room.teamId) {
```

Comment:
Instead of creating two branches of logic, it'd be nicer to unify the common logic and doing an extra call if `room.teamId`.

Reply:
True!